color.go

```diff
@@ -312,6 +313,52 @@ func boolPtr(v bool) *bool {
 	return &v
 }
 
+// colorsCache is used to reduce the count of created Color objects and
+// allows to reuse already created objects with required Attribute.
+var colorsCache = make(map[Attribute]*Color)
+
+// m is mutex for locking the access to cache map.
+var m = new(sync.Mutex)
+
+func getCachedColor(p Attribute) *Color {
+	m.Lock()
+	c, ok := colorsCache[p]
+	m.Unlock()
+	if !ok {
+		c = New(p)
+		m.Lock()
+		colorsCache[p] = c
+		m.Unlock()
```

Comment:
I know `defer` might be not so performant, but until we have a performance problem I prefer readable solution. So can you change this please to:

```
colorsCacheMu.Lock()
defer colorsCacheMu.Unlock()
``